test/e2e/scrapeconfig_test.go

```diff
@@ -0,0 +1,263 @@
+	sc := framework.MakeBasicScrapeConfig(ns, "scrape-config")
+	sc.Spec.StaticConfigs = []monitoringv1alpha1.StaticConfig{
+		{
+			Targets: []monitoringv1alpha1.Target{"target1:9090", "target2:9090"},
+		},
+	}
+	_, err = framework.MonClientV1alpha1.ScrapeConfigs(ns).Create(context.Background(), sc, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	// Check that the targets appear in Prometheus
+	err = wait.PollUntilContextTimeout(context.Background(), 5*time.Second, 2*time.Minute, false, func(ctx context.Context) (bool, error) {
+		targets, pollErr := framework.GetActiveTargets(ctx, ns, "prometheus-operated")
+		if pollErr != nil {
+			return false, nil
+		}
+
+		if len(targets) != 2 {
+			return false, nil
+		}
+
+		return true, nil
+	})
+	require.NoError(t, err)
+
+	// 2. Update the ScrapeConfig and add a target. Then, check that 3 targets appear in Prometheus.
+	sc, err = framework.GetScrapeConfig(context.Background(), ns, "scrape-config")
+	require.NoError(t, err)
+
+	sc.Spec.StaticConfigs = []monitoringv1alpha1.StaticConfig{
+		{
+			Targets: []monitoringv1alpha1.Target{"target1:9090", "target2:9090", "target3:9090"},
+		},
+	}
+
+	_, err = framework.UpdateScrapeConfig(context.Background(), ns, sc)
+	require.NoError(t, err)
+
+	// Check that the targets appear in Prometheus
+	err = wait.PollUntilContextTimeout(context.Background(), 5*time.Second, 5*time.Minute, false, func(ctx context.Context) (bool, error) {
+		targets, pollErr := framework.GetActiveTargets(ctx, ns, "prometheus-operated")
+		if pollErr != nil {
+			return false, nil
+		}
+
+		if len(targets) != 3 {
+			return false, nil
+		}
+
+		return true, nil
+	})
+	require.NoError(t, err)
+
+	// 3. Remove the ScrapeConfig and check that the targets disappear in Prometheus
+	err = framework.MonClientV1alpha1.ScrapeConfigs(ns).Delete(context.Background(), "scrape-config", metav1.DeleteOptions{})
+	require.NoError(t, err)
+
+	// Check that the targets disappeared in Prometheus
+	err = wait.PollUntilContextTimeout(context.Background(), 5*time.Second, 5*time.Minute, false, func(ctx context.Context) (bool, error) {
+		targets, pollErr := framework.GetActiveTargets(ctx, ns, "prometheus-operated")
+		if pollErr != nil {
+			return false, nil
+		}
+
+		if len(targets) != 0 {
+			return false, nil
+		}
+
+		return true, nil
+	})
+	require.NoError(t, err)
+}
+
+// testPromOperatorStartsWithoutScrapeConfigCRD deletes the ScrapeConfig CRD from the cluster and then starts
+// prometheus-operator to check that it doesn't crash.
+func testPromOperatorStartsWithoutScrapeConfigCRD(t *testing.T) {
+	skipPrometheusTests(t)
+
+	testCtx := framework.NewTestCtx(t)
+	defer testCtx.Cleanup(t)
+	ns := framework.CreateNamespace(context.Background(), t, testCtx)
+	framework.SetupPrometheusRBAC(context.Background(), t, testCtx, ns)
+
+	err := framework.DeleteCRD(context.Background(), "scrapeconfigs.monitoring.coreos.com")
+	require.NoError(t, err)
```

Comment:
Hey, looks like the scrapeConfig CRD was never installed in the first place 😅 

https://github.com/prometheus-operator/prometheus-operator/blob/820c2a1556053f5bd8393107c72a5ed2fc4f1e8d/test/framework/framework.go#L320-L327

I believe this snippet should be installing ScrapeConfig instead of PrometheusAgent

Reply:
Actually it is installing, just the function that checks if it was installed correctly is incorrect

Reply:
@ArthurSens I'm not sure I follow. Do you mean that `WaitForCRDReady` is not working as intended?

Reply:
Yes, because you're checking for PrometheusAgent instead of ScrapeConfig 😅 

Probably just a copy paste mistake

Reply:
oooooh indeed 😅 ``` 		return f.MonClientV1alpha1.PrometheusAgents(v1.NamespaceAll).List(ctx, opts) ```

Thanks!